[git-webkit checkout] Throws exception when branch is not in local repo

briannafan opened the pull request:
#### c6f8783a6157f999405149e4d12c945aac6430f5
<pre>
[git-webkit checkout] Throws exception when branch is not in local repo
<a href="https://bugs.webkit.org/show_bug.cgi?id=263030">https://bugs.webkit.org/show_bug.cgi?id=263030</a>
rdar://problem/116823751

Reviewed by NOBODY (OOPS!).

Handles commit not found exception when checking out a new branch.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git.checkout):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c6f8783a6157f999405149e4d12c945aac6430f5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21924 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/22145 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22991 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23806 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20303 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/22169 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/26391 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22456 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21388 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/22154 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/21788 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/19011 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24658 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/22104 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18917 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/19865 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/26133 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19946 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/20087 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23996 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/20527 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17490 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/21736 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19876 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/24083 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2723 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20477 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
```diff
@@ -1100,34 +1100,37 @@ def checkout(self, argument, prune=None, prompt=False):
-                    [self.executable(), 'checkout'] + ['-B', argument, remote_path] + log_arg,
+            try:
+                remote_path = '{}/{}'.format(branch_remote, argument)
+                local_head = self.commit(branch=argument, include_log=False, include_identifier=False)
```
Can we make this `try/catch` much smaller? I'm pretty sure it's just this line right here that needs it.

In general, `try/catch` in Python is a bit dangerous to apply this broadly because it can, for example, actually surpress syntax errors.

briannafan committed:
[git-webkit checkout] Throws exception when branch is not in local repo
https://bugs.webkit.org/show_bug.cgi?id=263030
rdar://problem/116823751

Reviewed by Jonathan Bedard.

Handles commit not found exception when checking out a new branch.

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git.checkout):

Canonical link: https://commits.webkit.org/269218@main
